src/components/show-products/index.js

```diff
@@ -1,25 +1,71 @@
-import React from 'react';
+import { doc, setDoc, serverTimestamp } from '@firebase/firestore';
+import { calculateEstimate } from '@the-collab-lab/shopping-list-utils';
 import { useHistory } from 'react-router';
-import Button from '../button';
-import { updatePurchaseDate } from '../../utils/firebaseUtils';
+import { db } from '../../lib/firebase';
+
+//Hooks
 import { useProducts } from '../../hooks/useProducts';
 import {
   useLocalStorage,
   LOCAL_STORAGE_LIST_TOKEN,
 } from '../../hooks/useLocalStorage';
+
+//Utils
+import { updatePurchaseDate } from '../../utils/firebaseUtils';
 import { TimeFrames } from '../../utils/timeFrames';
-import { compareDates } from '../../utils/compareDates';
+import {
+  diffBetweenTodayAndDate,
+  ONE_DAY,
+} from '../../utils/diffBetweenTodayAndDate';
+import { daysSinceLastTransaction } from '../../utils/daysSinceLastTransaction';
+import { findProductById } from '../../utils/findProductById';
 import ContentContainer from '../content-container';
+import Button from '../button';
+
 import './styles.css';
 
 export const ShowProducts = () => {
   const { products, loading } = useProducts();
   const { push } = useHistory();
   const { storedValue } = useLocalStorage(LOCAL_STORAGE_LIST_TOKEN);
   const timeFrames = TimeFrames;
+  const one_day = ONE_DAY;
 
-  const handleOnChange = (productID) => {
+  const handleOnChange = (event, productID) => {
     updatePurchaseDate(productID, storedValue);
+
+    const item = findProductById(products, productID);
+    const days = daysSinceLastTransaction(item, diffBetweenTodayAndDate());
```

Comment:
I think you forgot to pass some parameters to `diffBetweenTodayAndDate` function 👀 🙏 . It gave an error when I wanted to buy some product.

Reply:
I wanted to use that function as a callback, however when debugging I saw that the function is not sent as a callback, so I have no idea how to implement it :rofl: in this n commit [a0b55cf](https://github.com/the-collab-lab/tcl-34-smart-shopping-list/pull/25/commits/a0b55cf4f42dbff39b0d2701bb8d1c5679493368) I removed the function and returned to its previous config so the ternary operator runs there instead of using the function daysSinceLastTransaction